crates/services/executor/src/ports.rs

```diff
@@ -54,6 +66,53 @@ impl MaybeCheckedTransaction {
     }
 }
 
+pub trait TransactionExt {
+    fn max_gas(&self, consensus_params: &ConsensusParameters) -> ExecutorResult<u64>;
+}
+
```

Comment:
Nit: Since this is a one-method trait it would be easier to read the intention of it if it was named after the function: `pub trait MaxGas`.